### PR TITLE
RTE: Use <del> for strikeout

### PR DESCRIPTION
**Pull request by lukebarnard1**
We've swapped to commonmark which uses <del> for strikeout, so make the RTE use that. 

As a bonus, interpret ~~ as <del>.

land https://github.com/matrix-org/matrix-react-sdk/pull/1144 first

### CODE REVIEW

**Comment by t3chguy** [comment]
Might make sense to base this on `rte-fixes2`?

**Comment by t3chguy** [comment]
9404dd3 LGTM

**Comment by t3chguy** [comment]
Tests need updating: `PhantomJS 2.1.1 (Linux 0.0.0) MessageComposerInput should not strike-through ~~~ in Markdown mode FAILED`

File: src/Markdown.js [hunk 1]
```diff
@@ -62,6 +62,9 @@ function is_multi_line(node) {
  */
 export default class Markdown {
     constructor(input) {
+        // Support GH-style strikeout
+        input = input.replace(/~~(.*?)~~/g, '<del>$1</del>');
```
Comment (dbkr) [inline src/Markdown.js]: I thought we'd decided to not support this? Although I don't fully understand what context this is being use din

**Pull request merged**